acceptance-tests/src/test/java/tech/pegasys/signers/dsl/hashicorp/HashicorpNode.java

```diff
@@ -32,29 +37,37 @@ private HashicorpNode(final DockerClient dockerClient) {
   }
 
   private HashicorpNode(
-      final DockerClient dockerClient,
-      final HashicorpVaultDockerCertificate hashicorpVaultDockerCertificate) {
+      final DockerClient dockerClient, final SelfSignedCertificate tlsServerCertificate) {
     this.dockerClient = dockerClient;
-    this.hashicorpVaultDockerCertificate = hashicorpVaultDockerCertificate;
+    this.tlsServerCertificate = Optional.ofNullable(tlsServerCertificate);
   }
 
   public static HashicorpNode createAndStartHashicorp(
       final DockerClient dockerClient, final boolean withTls) {
-    final HashicorpNode hashicorpNode =
-        withTls
-            ? new HashicorpNode(dockerClient, HashicorpVaultDockerCertificate.create())
-            : new HashicorpNode(dockerClient);
-    hashicorpNode.start();
-    return hashicorpNode;
+    try {
+      final SelfSignedCertificate selfSignedCertificate = SelfSignedCertificate.generate();
```

Comment:
nit: there is no point in generate if withTls is false. May I suggest:
~~~
      final HashicorpNode hashicorpNode =
          withTls
              ? new HashicorpNode(dockerClient, SelfSignedCertificate.generate())
              : new HashicorpNode(dockerClient);
~~~

Reply:
nice.